sup3r/qa/stats.py

```diff
@@ -0,0 +1,494 @@
+        """Get stats for wind fields
+
+        Parameters
+        ----------
+        u: ndarray
+            Longitudinal velocity component
+            (lat, lon, temporal)
+        v : ndarray
+            Latitudinal velocity component
+            (lat, lon, temporal)
+
+        Returns
+        -------
+        stats : dict
+            Dictionary of stats for wind fields
+        """
+        v_dist = velocity_gradient_dist(u)
+        vort_dist = vorticity_dist(u, v)
+        rr_dist = ramp_rate_dist(u, v)
+        tke_ts = tke_series(u, v)
+        tke_avg = tke_spectrum(np.mean(u, axis=-1),
+                               np.mean(v, axis=-1))
+        return {'velocity_gradient': v_dist, 'vorticity': vort_dist,
+                'ramp_rate': rr_dist, 'tke_ts': tke_ts, 'tke_avg': tke_avg}
+
+    def get_height_stats(self, height):
+        """Get stats for high and low resolution wind fields
+
+        Parameters
+        ----------
+        height : int
+            Height in meters for requested U/V fields
+
+        Returns
+        -------
+        low_res : dict
+            Dictionary of stats for low resolution wind fields
+        high_res : dict
+            Dictionary of stats for high resolution wind fields
+        interp : dict
+            Dictionary of stats for spatiotemporally interpolated wind fields
+        """
+        u_hr, v_hr = self.get_uv_out(height)
+        uidx, vidx = self.feature_indices(height)
+        u_lr = self.source_handler.data[..., self.lr_t_slice, uidx]
+        v_lr = self.source_handler.data[..., self.lr_t_slice, vidx]
+        u_itp = st_interp(u_lr, self.s_enhance, self.t_enhance)
+        v_itp = st_interp(v_lr, self.s_enhance, self.t_enhance)
+
+        low_res = self.get_wind_stats(u_lr, v_lr)
+        high_res = self.get_wind_stats(u_hr, v_hr)
+        interp = self.get_wind_stats(u_itp, v_itp)
+        return low_res, high_res, interp
+
+    def run(self):
+        """Go through all datasets and get the error for the re-coarsened
+        synthetic minus the true low-res source data.
+
+        Returns
+        -------
+        errors : dict
+            Dictionary of errors, where keys are the feature names, and each
+            value is an array with shape (space1, space2, time) that represents
+            the re-coarsened synthetic data minus the source true low-res data
+        """
+
+        stats = {}
+        for idf, height in enumerate(self.heights):
+            logger.info('Running WindStats on height {} of {}: "{}m"'
+                        .format(idf + 1, len(self.heights), height))
+            lr_stats, hr_stats, interp = self.get_height_stats(height)
+
+            stats[f'lr_{height}m'] = lr_stats
+            stats[f'hr_{height}m'] = hr_stats
+            stats[f'interp_{height}m'] = interp
+
+        if self.qa_fp is not None:
+            self.export(self.qa_fp, stats)
+        logger.info('Finished Sup3rWindStats run method.')
+
+        return stats
```

Comment:
We should probably test our stats module... it's not really QA if we don't is it? Unit tests on the qa utilities would be best.

Reply:
Yeah I was waiting for the final collection on wtk-led to finish. I had to rerun due to some of those same issues I showed you yesterday, with the empty chunks. I think I traced this back to n_writes splitting the file list such that the spatial domains for each file list chunk didnt match.